test/com/google/enterprise/adaptor/filenet/FileNetAdaptorTest.java

```diff
@@ -379,49 +379,57 @@ public void testInit_objectFactory_missing() throws Exception {
   }
 
   @Test
-  public void testInit_displayUrl_bare() throws Exception {
-    config.overrideKey("filenet.displayUrl", "http://localhost");
-    config.overrideKey("filenet.objectStore", "ObjectStore");
+  public void testInit_displayUrl_default() throws Exception {
     adaptor.init(context);
-    String expected = "http://localhost/getContent?objectStoreName=ObjectStore"
-        + "&objectType=document&versionStatus=1&vsId=";
-    assertEquals(expected, getConfigOptions().getDisplayUrl());
+    Id guid = new Id("{AAAAAAAA-0000-0000-0000-000000000001}");
+    Id vsId = new Id("{AAAAAAAA-0000-0000-0000-000000000002}");
+    String expected = "http://localhost/WorkplaceXT/getContent"
+        + "?objectStoreName=ObjectStore"
+        + "&objectType=document&versionStatus=1&vsId=" + percentEscape(vsId);
+    assertEquals(expected,
+        getConfigOptions().getDisplayUrl(guid, vsId).toString());
   }
 
   @Test
-  public void testInit_displayUrl_getContent() throws Exception {
-    config.overrideKey("filenet.displayUrl", "http://localhost/getContent");
-    config.overrideKey("filenet.objectStore", "ObjectStore");
+  public void testInit_displayUrl_relative() throws Exception {
+    config.overrideKey("filenet.contentEngineUrl", "https://localhost:8080/");
+    config.overrideKey("filenet.displayUrlPattern", "getContent?os={2}&vs={1}");
     adaptor.init(context);
-    String expected = "http://localhost/getContent?objectStoreName=ObjectStore"
-        + "&objectType=document&versionStatus=1&vsId=";
-    assertEquals(expected, getConfigOptions().getDisplayUrl());
+    Id guid = new Id("{AAAAAAAA-0000-0000-0000-000000000001}");
+    Id vsId = new Id("{AAAAAAAA-0000-0000-0000-000000000002}");
+    String expected = "https://localhost:8080/getContent?os=ObjectStore&vs="
+        + percentEscape(vsId);
+    assertEquals(expected,
+        getConfigOptions().getDisplayUrl(guid, vsId).toString());
   }
 
   @Test
-  public void testInit_displayUrl_getContentSlash() throws Exception {
-    config.overrideKey("filenet.displayUrl", "http://localhost/getContent/");
-    config.overrideKey("filenet.objectStore", "ObjectStore");
+  public void testInit_displayUrl_absolute() throws Exception {
+    config.overrideKey("filenet.displayUrlPattern",
+        "https://localhost:8080/getContent?os=ObjStore&vs={1}&guid={0}");
```

Comment:
Maybe s{getContent?os=ObjStore}{whatever?objStore=NotObjectStore}, just to signal that WorkplaceXT URLs are not required?